zhmccli/_cmd_partition.py

```diff
@@ -703,3 +746,38 @@ def cmd_partition_console(cmd_ctx, cpc_name, partition_name, options):
         part_console(cmd_ctx.session, partition, refresh, logger)
     except zhmcclient.Error as exc:
         raise click.ClickException("%s: %s" % (exc.__class__.__name__, exc))
+
+
+def cmd_partition_mount_iso(cmd_ctx, cpc_name, partition_name, options):
+
+    client = zhmcclient.Client(cmd_ctx.session)
+    partition = find_partition(cmd_ctx, client, cpc_name, partition_name)
+
+    image_file = options['imagefile']
+    image_fp = open(image_file, 'rb')
+    path, image_name = os.path.split(image_file)
+    partition.mount_iso_image(image_fp, image_name, options['imageinsfile'])
+    if options['boot']:
+        partition.update_properties({'boot-device': 'iso-image'})
+    cmd_ctx.spinner.stop()
+    click.echo('ISO image %s has been mounted to Partition %s.' %
+               (image_name, partition.name))
+
+
+def cmd_partition_unmount_iso(cmd_ctx, cpc_name, partition_name, options):
+
+    client = zhmcclient.Client(cmd_ctx.session)
+    partition = find_partition(cmd_ctx, client, cpc_name, partition_name)
+
+    partition.pull_full_properties()
+    image_name = partition.get_property('boot-iso-image-name')
+    if options['boot']:
+        partition.update_properties({'boot-device': ''})
```

Comment:
Move the unsetting of the `boot-device` property after the unmounting, in case that one fails.